### PR TITLE
feat: path globbing

### LINKED ISSUE #14: Feature: Paths should be globs
Using a singular entrypoint which is only a folder is limiting.  I would like the ability to pick and choose which less file(s) are being compiled without compiling all of them.   Using a [library like glob](https://github.com/isaacs/node-glob) will make it much easier to specify multiple files via patterns and recursiveness.

### PR DESCRIPTION
**Pull request by curbengh**
Related to #14 

globbing only applies to hexo-processed files; e.g. doesn't apply to node_modules/ folder.

### CODE REVIEW

File: lib/renderer.js [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
-  const paths = config.paths.map(filepath => {
-    return join(cwd, filepath); // assuming paths are relative from the root of the project
-  });
+  let { paths } = config;
+  const routeList = route.list();
+
+  if (!paths) paths = [];
+
+  paths = micromatch(routeList, paths);
+  paths = paths.map(path => join(cwd, dirname(path)));
```
Comment (curbengh) [inline lib/renderer.js]: Don't know whether this works in Windows.

**Comment by SukkaW** [comment]
@curbengh micromatch has a `is-glob`: https://github.com/micromatch/is-glob/

Could it be used to avoid Breaking Changes?

File: lib/renderer.js [hunk 2]
```diff
@@ -1,21 +1,28 @@
...
+  const { route, theme } = this;
 
-  const config = this.theme.config.less;
+  const config = theme.config.less;
   const cwd = process.cwd();
-  const paths = config.paths.map(filepath => {
-    return join(cwd, filepath); // assuming paths are relative from the root of the project
-  });
+  let { paths } = config;
+  const routeList = route.list();
```
Comment (curbengh) [inline lib/renderer.js]: Probably not the correct approach, as `paths:` usually refers to a folder outside of hexo processing (e.g. node_modules), which is not intended to copied into 'public/' folder.

**Comment by curbengh** [comment]
> @curbengh micromatch has a `is-glob`: https://github.com/micromatch/is-glob/
> 
> Could it be used to avoid Breaking Changes?

incorporated isGlob.

**Comment by curbengh** [comment]
#14 requires globbing outside of hexo-processed files, like node_modules/ folder, as it mentions node-glob.

I'm not so keen on node-glob because I prefer not to use the dated minimatch. micromatch has a replacement called [glob-fs](https://github.com/micromatch/glob-fs), but it's not a drop-in as it doesn't support windows.

**Comment by SukkaW** [comment]
Any ideas about non-glob support? For example, add suffix `**` or `/**` when processing the config.

File: lib/renderer.js [hunk 3]
```diff
@@ -1,21 +1,31 @@
...
+  let tmpPaths = config.paths;
+
+  if (!tmpPaths) tmpPaths = [];
+  else if (typeof tmpPaths === 'string') tmpPaths = [tmpPaths];
+
+  const paths = tmpPaths.filter(path => !micromatch.scan(path).isGlob)
+    .map(path => join(cwd, path)); // assuming paths are relative from the root of the project;
+  const glob = tmpPaths.filter(path => micromatch.scan(path).isGlob);
+  const match = micromatch(routeList, glob).map(path => join(cwd, dirname(path)));
```
Comment (curbengh) [inline lib/renderer.js]: > Any ideas about non-glob support?

forgot about the non-glob. In retrospect, I don't think it's necessary to filter out glob patterns, can pass the config `config.paths` directly to micromatch.

**Comment by curbengh** [comment]
Since micromatch is included, can implement `exclude:` option too (in a separate PR).

**Comment by curbengh** [comment]
Rebased

**Pull request merged**